server/src/main/java/org/opensearch/cluster/routing/WeightedRoutingService.java

```diff
@@ -189,26 +192,41 @@ public void verifyAwarenessAttribute(String attributeName) {
 
     private void ensureWeightsSetForAllDiscoveredAndForcedAwarenessValues(ClusterState state, ClusterPutWeightedRoutingRequest request) {
```

Comment:
Do we have a test written for this case?

Reply:
We have a test for this - 
```
public void testAddWeightedRoutingFailsWhenWeightsNotSetForAllDiscoveredZones() throws InterruptedException {
        ClusterPutWeightedRoutingRequestBuilder request = new ClusterPutWeightedRoutingRequestBuilder(
            client,
            ClusterAddWeightedRoutingAction.INSTANCE
        );
        Map<String, Double> weights = Map.of("zone_A", 1.0, "zone_C", 1.0);
        WeightedRouting weightedRouting = new WeightedRouting("zone", weights);
        request.setWeightedRouting(weightedRouting);
        final CountDownLatch countDownLatch = new CountDownLatch(1);
        final AtomicReference<Exception> exceptionReference = new AtomicReference<>();
        ActionListener<ClusterStateUpdateResponse> listener = new ActionListener<ClusterStateUpdateResponse>() {
            @Override
            public void onResponse(ClusterStateUpdateResponse clusterStateUpdateResponse) {
                countDownLatch.countDown();
            }

            @Override
            public void onFailure(Exception e) {
                exceptionReference.set(e);
                countDownLatch.countDown();
            }
        };
        weightedRoutingService.registerWeightedRoutingMetadata(request.request(), listener);
        assertTrue(countDownLatch.await(30, TimeUnit.SECONDS));
        MatcherAssert.assertThat("Expected onFailure to be called", exceptionReference.get(), notNullValue());
        MatcherAssert.assertThat(exceptionReference.get(), instanceOf(UnsupportedWeightedRoutingStateException.class));
        MatcherAssert.assertThat(
            exceptionReference.get().getMessage(),
            containsString("weight for [zone_B] is not set and it is part of forced awareness value or a routing node has this attribute.")
        );
    }
```